RecoMET/METAlgorithms/src/HcalHaloAlgo.cc

```diff
@@ -54,83 +53,73 @@ HcalHaloData HcalHaloAlgo::Calculate(const CaloGeometry& TheCaloGeometry, edm::H
       MaxTimeHits[i] = 0.;
     }
   
-  for( HBHERecHitCollection::const_iterator hit = TheHBHERecHits->begin() ; hit != TheHBHERecHits->end() ; hit++ )
-    {
-      HcalDetId id = HcalDetId(hit->id());                                                                                                    
-      switch ( id.subdet() )                                                                                         
-	{      
-	case HcalBarrel:                                                                           
-	  if(hit->energy() < HBRecHitEnergyThreshold )continue;
-	  break;                                                                                                                  
-	case HcalEndcap:                                                                                          
-	  if(hit->energy() < HERecHitEnergyThreshold ) continue;
-	  break;
-	default:
-	  continue;
-	}
-      
-      int iEta = id.ieta();
-      int iPhi = id.iphi();
-      if(iPhi < 73 && TMath::Abs(iEta) < 23 )
-	{ 
-	  SumE[iPhi]+= hit->energy();
-	  NumHits[iPhi] ++;
+  for (const auto & hit : (HBHERecHitCollection)(*TheHBHERecHits)) {
+    HcalDetId id = HcalDetId(hit.id());
+    switch ( id.subdet() ) {      
+    case HcalBarrel:
+      if(hit.energy() < HBRecHitEnergyThreshold )continue;
+      break;
+    case HcalEndcap:
+      if(hit.energy() < HERecHitEnergyThreshold ) continue;
+      break;
+    default:
+      continue;
+    }
+    
+    int iEta = id.ieta();
+    int iPhi = id.iphi();
+    if(iPhi < 73 && std::abs(iEta) < 23 ) { 
```

Comment:
It is a pitty that those magic numbers are not defined anywhere centrally (at least, I was not able to find them), because they are hardcoded rather often in the HCal code throughout CMSSW
Not requested for this PR, though, unless they are already defined somewhere and it is just a matter of linking them here

Reply:
I believe the cut on iEta comes from some specific analysis to get the halo. the cut on iPhi is redundant unless there is some overlap with HGCal code. But I am not the author of the analysis who defined the Halo suppression code.